pkg/tcpserver/tcp_server.go

```diff
@@ -93,6 +95,9 @@ func NewTCPServer(address string, credentials *security.Credential) (TCPServer,
 
 // Run runs the mux. The mux has to be running to accept connections.
 func (s *tcpServerImpl) Run(ctx context.Context) error {
+	defer func() {
+		_ = s.rootListener.Close()
```

Comment:
`s.rootListener.Close()` was called in `Run` and `Close`.

why do we have to call this method here ?

Reply:
`s.rootListener.Close()` provides a reliable way to tell downstream components such as the HTTP API server to exit.